github/DiscussionCommentBase.py

```diff
@@ -0,0 +1,117 @@
+############################ Copyrights and license ############################
+#                                                                              #
+# Copyright 2024 Enrico Minack <github@enrico.minack.dev>                      #
+#                                                                              #
+# This file is part of PyGithub.                                               #
+# http://pygithub.readthedocs.io/                                              #
+#                                                                              #
+# PyGithub is free software: you can redistribute it and/or modify it under    #
+# the terms of the GNU Lesser General Public License as published by the Free  #
+# Software Foundation, either version 3 of the License, or (at your option)    #
+# any later version.                                                           #
+#                                                                              #
+# PyGithub is distributed in the hope that it will be useful, but WITHOUT ANY  #
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS    #
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more #
+# details.                                                                     #
+#                                                                              #
+# You should have received a copy of the GNU Lesser General Public License     #
+# along with PyGithub. If not, see <http://www.gnu.org/licenses/>.             #
+#                                                                              #
+################################################################################
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any
+
+import github.GithubObject
+import github.NamedUser
+from github.GithubObject import Attribute, CompletableGithubObject, NotSet
+
+
+class DiscussionCommentBase(CompletableGithubObject):
+    """
+    This class represents a the shared attributes between RepositoryDiscussionComment and TeamDiscussionComment
+    https://docs.github.com/en/graphql/reference/objects#discussioncomment
+    https://docs.github.com/de/rest/teams/discussion-comments
+    """
+
+    def _initAttributes(self) -> None:
+        self._author: Attribute[github.NamedUser.NamedUser] = NotSet
+        self._body: Attribute[str] = NotSet
+        self._body_html: Attribute[str] = NotSet
+        self._created_at: Attribute[datetime] = NotSet
+        self._html_url: Attribute[str] = NotSet
+        self._last_edited_at: Attribute[datetime] = NotSet
+        self._node_id: Attribute[str] = NotSet
+        self._updated_at: Attribute[datetime] = NotSet
+        self._url: Attribute[str] = NotSet
+
+    def __repr__(self) -> str:
+        return self.get__repr__({"node_id": self._node_id.value})
+
+    @property
+    def author(self) -> github.NamedUser.NamedUser:
```

Comment:
```suggestion
    def author(self) -> github.NamedUser.NamedUser | None:
```

The author could be `None` if the user was deleted.